Quorum failures

**com.hazelcast.quorum.map.MapReadQuorumTest**
```java
java.lang.AssertionError: expected:<3> but was:<4>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.quorum.PartitionedCluster.splitFiveMembersThreeAndTwo(PartitionedCluster.java:136)
	at com.hazelcast.quorum.PartitionedCluster.partitionFiveMembersThreeAndTwo(PartitionedCluster.java:62)
	at com.hazelcast.quorum.map.MapReadQuorumTest.initialize(MapReadQuorumTest.java:66)
```
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/new-lab-fast-pr/8395/testReport/junit/com.hazelcast.quorum.map/MapReadQuorumTest/com_hazelcast_quorum_map_MapReadQuorumTest/
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/new-lab-fast-pr/8391/testReport/junit/com.hazelcast.quorum.map/MapReadQuorumTest/com_hazelcast_quorum_map_MapReadQuorumTest/

**com.hazelcast.client.quorum.ClientMapReadWriteQuorumTest**
```java
java.lang.AssertionError: expected:<3> but was:<4>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.quorum.PartitionedCluster.splitFiveMembersThreeAndTwo(PartitionedCluster.java:136)
	at com.hazelcast.quorum.PartitionedCluster.partitionFiveMembersThreeAndTwo(PartitionedCluster.java:62)
	at com.hazelcast.client.quorum.ClientMapReadWriteQuorumTest.initialize(ClientMapReadWriteQuorumTest.java:80)
```
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/new-lab-fast-pr/8379/testReport/junit/com.hazelcast.client.quorum/ClientMapReadWriteQuorumTest/com_hazelcast_client_quorum_ClientMapReadWriteQuorumTest/

**com.hazelcast.client.quorum.cache.ClientCacheReadQuorumTest.testGetAsyncOperationSuccessfulWhenQuorumSizeMet**
```java
java.util.concurrent.ExecutionException: com.hazelcast.quorum.QuorumException: Cluster quorum failed, quorum minimum size: 3, current size: 2
	at com.hazelcast.quorum.impl.QuorumImpl.newQuorumException(QuorumImpl.java:167)
	at com.hazelcast.quorum.impl.QuorumImpl.ensureQuorumPresent(QuorumImpl.java:158)
	at com.hazelcast.quorum.impl.QuorumServiceImpl.ensureQuorumPresent(QuorumServiceImpl.java:127)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.ensureQuorumPresent(OperationRunnerImpl.java:245)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:186)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:397)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:424)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:536)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:521)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:492)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:202)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59)
	at com.hazelcast.client.impl.protocol.task.AbstractPartitionMessageTask.processMessage(AbstractPartitionMessageTask.java:64)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:123)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:103)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:154)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:127)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:102)
	at ------ submitted from ------.(Unknown Source)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:120)
	at com.hazelcast.spi.impl.AbstractInvocationFuture$1.run(AbstractInvocationFuture.java:243)
	at com.hazelcast.client.impl.protocol.task.AbstractPartitionMessageTask.execute(AbstractPartitionMessageTask.java:78)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.unblock(AbstractInvocationFuture.java:239)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.andThen(AbstractInvocationFuture.java:215)
	at com.hazelcast.client.impl.protocol.task.AbstractPartitionMessageTask.processMessage(AbstractPartitionMessageTask.java:69)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:123)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:103)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:154)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:127)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:102)
	at ------ submitted from ------.(Unknown Source)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:93)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:33)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:183)
	at com.hazelcast.client.util.ClientDelegatingFuture.get(ClientDelegatingFuture.java:151)
	at com.hazelcast.client.util.ClientDelegatingFuture.get(ClientDelegatingFuture.java:137)
	at com.hazelcast.client.quorum.cache.ClientCacheReadQuorumTest.testGetAsyncOperationSuccessfulWhenQuorumSizeMet(ClientCacheReadQuorumTest.java:173)
```
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/new-lab-fast-pr/8372/testReport/junit/com.hazelcast.client.quorum.cache/ClientCacheReadQuorumTest/testGetAsyncOperationSuccessfulWhenQuorumSizeMet/

**com.hazelcast.client.quorum.cache.ClientCacheReadQuorumTest.testContainsOperationSuccessfulWhenQuorumSizeMet**
```java
com.hazelcast.quorum.QuorumException: Cluster quorum failed, quorum minimum size: 3, current size: 2
	at com.hazelcast.quorum.impl.QuorumImpl.newQuorumException(QuorumImpl.java:167)
	at com.hazelcast.quorum.impl.QuorumImpl.ensureQuorumPresent(QuorumImpl.java:158)
	at com.hazelcast.quorum.impl.QuorumServiceImpl.ensureQuorumPresent(QuorumServiceImpl.java:127)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.ensureQuorumPresent(OperationRunnerImpl.java:245)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:186)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:397)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:424)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:536)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:521)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:492)
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:202)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:59)
	at com.hazelcast.client.impl.protocol.task.AbstractPartitionMessageTask.processMessage(AbstractPartitionMessageTask.java:64)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:123)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:103)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:154)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:127)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:102)
	at ------ submitted from ------.(Unknown Source)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:120)
	at com.hazelcast.spi.impl.AbstractInvocationFuture$1.run(AbstractInvocationFuture.java:243)
	at com.hazelcast.client.impl.protocol.task.AbstractPartitionMessageTask.execute(AbstractPartitionMessageTask.java:78)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.unblock(AbstractInvocationFuture.java:239)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.andThen(AbstractInvocationFuture.java:215)
	at com.hazelcast.client.impl.protocol.task.AbstractPartitionMessageTask.processMessage(AbstractPartitionMessageTask.java:69)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:123)
	at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:103)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:154)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:127)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:102)
	at ------ submitted from ------.(Unknown Source)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:93)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:33)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:155)
	at com.hazelcast.client.cache.impl.AbstractClientCacheProxyBase.invoke(AbstractClientCacheProxyBase.java:232)
	at com.hazelcast.client.cache.impl.ClientCacheProxy.containsKeyInternal(ClientCacheProxy.java:108)
	at com.hazelcast.client.cache.impl.ClientCacheProxy.containsKey(ClientCacheProxy.java:103)
	at com.hazelcast.client.quorum.cache.ClientCacheReadQuorumTest.testContainsOperationSuccessfulWhenQuorumSizeMet(ClientCacheReadQuorumTest.java:138)
```
https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/new-lab-fast-pr/8372/testReport/junit/com.hazelcast.client.quorum.cache/ClientCacheReadQuorumTest/testContainsOperationSuccessfulWhenQuorumSizeMet/

https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/8410/

```
java.lang.AssertionError: expected:<3> but was:<4>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at com.hazelcast.quorum.PartitionedCluster.splitFiveMembersThreeAndTwo(PartitionedCluster.java:137)
	at com.hazelcast.client.quorum.cache.ClientCacheReadQuorumTest.initialize(ClientCacheReadQuorumTest.java:86)
```

com.hazelcast.client.quorum.cache.ClientCacheReadQuorumTest

Failed again in the PR builder:

**com.hazelcast.quorum.cache.CacheReadWriteQuorumTest**
```java
java.lang.AssertionError: Cluster size is not correct. Expected: 3 Actual: 4
	at org.junit.Assert.fail(Assert.java:88)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:859)
	at com.hazelcast.test.HazelcastTestSupport$12.run(HazelcastTestSupport.java:878)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:992)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:874)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:869)
	at com.hazelcast.quorum.PartitionedCluster.splitFiveMembersThreeAndTwo(PartitionedCluster.java:136)
	at com.hazelcast.quorum.PartitionedCluster.partitionFiveMembersThreeAndTwo(PartitionedCluster.java:69)
	at com.hazelcast.quorum.cache.CacheReadWriteQuorumTest.initialize(CacheReadWriteQuorumTest.java:83)
```
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/8427/testReport/junit/com.hazelcast.quorum.cache/CacheReadWriteQuorumTest/com_hazelcast_quorum_cache_CacheReadWriteQuorumTest/

**com.hazelcast.client.quorum.ClientMapWriteQuorumTest**
```
java.lang.AssertionError: Cluster size is not correct. Expected: 3 Actual: 4
	at org.junit.Assert.fail(Assert.java:88)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:859)
	at com.hazelcast.test.HazelcastTestSupport$12.run(HazelcastTestSupport.java:878)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:992)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:874)
	at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:869)
	at com.hazelcast.quorum.PartitionedCluster.splitFiveMembersThreeAndTwo(PartitionedCluster.java:136)
	at com.hazelcast.quorum.PartitionedCluster.partitionFiveMembersThreeAndTwo(PartitionedCluster.java:63)
	at com.hazelcast.client.quorum.ClientMapWriteQuorumTest.initialize(ClientMapWriteQuorumTest.java:80)
```
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/8430/testReport/junit/com.hazelcast.client.quorum/ClientMapWriteQuorumTest/com_hazelcast_client_quorum_ClientMapWriteQuorumTest/

--- a/hazelcast-client/src/test/java/com/hazelcast/client/quorum/QuorumTestUtil.java
+++ b/hazelcast-client/src/test/java/com/hazelcast/client/quorum/QuorumTestUtil.java
@@ -31,6 +31,7 @@ public class QuorumTestUtil {
         ClientConfig clientConfig = new ClientConfig();
         Address address = getNode(instance).address;
         clientConfig.getNetworkConfig().addAddress(address.getHost() + ":" + address.getPort());
+        clientConfig.getNetworkConfig().setSmartRouting(false);
         clientConfig.getGroupConfig().setName(instance.getConfig().getGroupConfig().getName());
         return clientConfig;
     }

